Adding alert when closing the editor with unsaved changes

etoledom opened the pull request:
Closes #714 

### Description

This PR adds an alert when the user tries to close the quick editor while there are unsaved changes.

The alert will be shown also in the "avatar & about" scope, while showing the avatar picker, if there are still unsaved changes in the about editor side.

![CleanShot 2025-05-08 at 08 13 09](https://github.com/user-attachments/assets/7fb3dbe5-1574-4be4-8955-f44094263434)


### Testing Steps

- Launch the demo app on the QE screen
- Select About Editor scope 
- Open the QE 
- Close the QE
 - It should close without warning 
- Open the QE and make some changes
- Close the QE 
  - An alert warning should appear
- Test both action buttons
- Repeat opening QE and making changes
- Save changes and close QE
  - No alert should be presented  
- Repeat opening the QE using the Avatar & About scope
- Make changes to the About section
- Switch to the Avatar picker section
- Close the QE
  - The alert should be presented.

etoledom committed:
Adding alert when closing the editor with unsaved changes

etoledom committed:
Format

etoledom commented on Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarPickerViewModel.swift:
```diff
@@ -381,25 +385,32 @@ class AvatarPickerViewModel: ObservableObject {
+            await avatars
+            await profile
+        }
     }
```
This solves a problem where switching between avatar picker and about editor would remove the unsaved changes on the about editor side.

pinarol commented:
I can just swipe down and close the sheet without the alert, don't you think we need to disable that too if there are unsaved changes? I expect swiping down to be used way more often than tapping the Done button.

etoledom committed:
Block dismissal of QE by dragging down if there are unsaved changes.

etoledom committed:
Intercepting swipe down dismissal of QE on UIKit

With this we can show the unsaved changes alert when the user tries to dismiss with a gesture.
So far only works presenting the QE from UIKit context.

etoledom committed:
Format

etoledom committed:
Merge branch 'trunk' into etoledom/detect-unsaved-changes

etoledom committed:
Fix merge conflicts

pinarol committed:
SwiftUI sheet dismiss detection (#743)

Co-authored-by: Eduardo Toledo <etoledom@icloud.com>

etoledom committed:
Remove unnecesary Equatable conformance

etoledom committed:
Merge branch 'etoledom/detect-unsaved-changes' of https://github.com/Automattic/Gravatar-SDK-iOS into etoledom/detect-unsaved-changes

etoledom committed:
Remove unused `updatedAboutInfoModel` property

etoledom committed:
Threshold modification

etoledom committed:
fix `onChange(of:)` usage

etoledom committed:
Remove extra condition to detect sheet dragged down

etoledom committed:
Fix crash on UIKit demo app

pinarol commented on Sources/GravatarUI/SwiftUI/QuickEditorModalPresentationModifier.swift:
```diff
@@ -24,6 +24,8 @@ struct QuickEditorModalPresentationModifier<ModalView: View>: ViewModifier, Moda
     @State private var prioritizeScrollOverResize: Bool = false
     @Environment(\.colorScheme) var colorScheme: ColorScheme
+    @State private var dismissAttempt: Bool = false
+    @State private var debounceWorkItem: DispatchWorkItem?
```
Seem like, I forgot to remove this var `debounceWorkItem`. It's left from an experiment.

pinarol reviewed (approved):
Looks good 🎉 We can remove that variable and merge I think.

etoledom committed:
Removing unused property
